### PR TITLE
Fix N+1

### PR DESCRIPTION
**Pull request by Thinus01**
### General requirements

- Made sure that there are no linter errors.
- Made sure that I used correct flow Gitflow.
- Made sure that I documented my work in a professional way.

### Project requirements

Integration Specs:

Using Capybara, I have written integration tests for each view in my project. I have organized the tests of each page in a way that makes sense for the flow of things. For example, one test checks for the page elements and another test checks for clicking links.

User Index Page:

I can see the usernames of all other users, their profile picture, and the number of posts each user has written. When I click on a user, I am redirected to that user's show page.

User Show Page:

I can see the user's profile picture, username, bio, and the number of posts the user has written. I can also see the user's first three posts and a button that lets me view all of a user's posts. When I click on a user's post, it redirects me to that post's show page. When I click to see all posts, it redirects me to the user's post's index page.

User Post Index Page:

I can see the user's profile picture, username, and the number of posts the user has written. I can also see a post's title, some of its body, and the first comments on the post. I can see how many comments and likes a post has. If there are more posts than fit on the view, I can see a section for pagination. When I click on a post, it redirects me to that post's show page.

Post Show Page:

I can see the post's title, who wrote the post, how many comments and likes it has, and the post body. I can also see the username and comment left by each commentor.

Here is my terminal before the fix
not_fix

Here is my terminal after the fix
n+1Fix

### CODE REVIEW

File: spec/system/posts_index_spec.rb [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+  it 'likes_counter must not be less than 1' do
+    subject.likes_counter = -1
+    expect(subject).to_not be_valid
+  end
+
+  it 'should return 5 recent comments' do
+    expect(subject.recent_comments).to eq(subject.comments.order(created_at: :asc).limit(5))
+  end
+end
```
Comment (julie-ify) [inline spec/system/posts_index_spec.rb]: - [ ] This test file does not look like a `feature` type test for the post index page. It is a model test file for Post item. Kindly write the correct tests and move this to its correct sub-directory.

- [ ] Your test needs to visit the require post index route and perform assertions based on the requirements.

File: spec/requests/posts_spec.rb [hunk 2]
```diff
@@ -1,36 +1,36 @@
-require 'rails_helper'
-RSpec.describe '/posts', type: :request do
-  let(:seeded_user) { User.first } # pick the first user from seed file
-  let(:seeded_post) { seeded_user.posts.first } # use the first post of seeded user for show action
+# require 'rails_helper'
+# RSpec.describe '/posts', type: :request do
+#   let(:seeded_user) { User.first } # pick the first user from seed file
+#   let(:seeded_post) { seeded_user.posts.first } # use the first post of seeded user for show action
```
Comment (julie-ify) [inline spec/requests/posts_spec.rb]: - [ ] Kindly remove commented code.

File: spec/requests/posts_spec.rb [hunk 3]
```diff
@@ -1,36 +1,36 @@
...
-    it 'contains the correct text' do
-      expect(response.body).to include('Here is a post POST for user X')
-    end
-  end
-end
+#     it 'contains the correct text' do
+#       expect(response.body).to include('Here is a post POST for user X')
+#     end
+#   end
+# end
```
Comment (julie-ify) [inline spec/requests/posts_spec.rb]: - [ ] Kindly remove commented code or remove this file if you do not intend using it.

File: spec/requests/users_spec.rb [hunk 4]
```diff
@@ -1,34 +1,34 @@
...
+#       expect(response).to be_successful
+#     end
+#     it 'renders the correct template' do
+#       expect(response).to render_template(:show)
+#     end
+#     it 'contains the correct text' do
+#       expect(response.body).to include('Here are the detials about User ABC')
+#     end
+#   end
+# end
```
Comment (julie-ify) [inline spec/requests/users_spec.rb]: - [ ] Kindly remove commented code or the file entirely if you do not intend using it.

**Review by julie-ify — CHANGES_REQUESTED** [review]
## Hi @Thinus01, @Fahim-Khan-P and @lawrahkonwea,

Good job so far!

## Highlights!
- Good feature branch name ✔️ 
- No linter errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 
- Good code organization and working app ✔️
- Descriptive README ✔️ 

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@julie-ify` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by addod19 — APPROVED** [review]
### Hello TEAM,

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

### HIGHLIGHTS
- [ ] Nice job fixing all the comments from the previous review :rocket: 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**